src/main/java/net/sf/jabref/gui/openoffice/OpenOfficePanel.java

```diff
@@ -322,6 +332,39 @@ public void actionPerformed(ActionEvent e) {
 
     }
 
+    private void exportEntries() {
+        try {
+            if (style == null) {
+                style = loader.getUsedStyle();
+            } else {
+                style.ensureUpToDate();
+            }
+
+            ooBase.updateSortedReferenceMarks();
+
+            List<BibDatabase> databases = getBaseList();
+            List<String> unresolvedKeys = ooBase.refreshCiteMarkers(databases, style);
+            BibDatabase newDatabase = ooBase.generateDatabase(databases);
+            if (!unresolvedKeys.isEmpty()) {
+                JOptionPane.showMessageDialog(frame,
+                        Localization.lang(
+                                "Your OpenOffice/LibreOffice document references the BibTeX key '%0', which could not be found in your current database.",
+                                unresolvedKeys.get(0)),
+                        Localization.lang("Unable to generate new database"), JOptionPane.ERROR_MESSAGE);
+            }
+
+            Defaults defaults = new Defaults(
+                    BibDatabaseMode.fromPreference(Globals.prefs.getBoolean(JabRefPreferences.BIBLATEX_DEFAULT_MODE)));
+
+            BibDatabaseContext databaseContext = new BibDatabaseContext(newDatabase, defaults);
+            this.frame.addTab(databaseContext, true);
+
+        } catch (Exception e) {
```

Comment:
The problem is that `refreshCiteMarkers` throws `Exception` because `refreshCiteMarkersInternal` throws `Exception` because `insertReferenceMark` throws `Exception` because `com.sun.star.lang.XMultiServiceFactory.createInstance` throws `Exception`. 

I have earlier tried to specify the exceptions in OO/LO where possible, but there are still some methods that only throw `Exception`. It would be possible to borrow the code from the update action:

``` java
                } catch (UndefinedCharacterFormatException ex) {
                    reportUndefinedCharacterFormat(ex);
                } catch (UndefinedParagraphFormatException ex) {
                    reportUndefinedParagraphFormat(ex);
                } catch (ConnectionLostException ex) {
                    showConnectionLostErrorMessage();
                } catch (IOException ex) {
                    JOptionPane.showMessageDialog(frame,
                            Localization
                                    .lang("You must select either a valid style file, or use one of the default styles."),
                            Localization.lang("No valid style file defined"), JOptionPane.ERROR_MESSAGE);
                    LOGGER.warn("Problem with style file", ex);
                } catch (BibEntryNotFoundException ex) {
                    JOptionPane.showMessageDialog(frame,
                            Localization.lang(
                                    "Your OpenOffice/LibreOffice document references the BibTeX key '%0', which could not be found in your current database.",
                                    ex.getBibtexKey()),
                            Localization.lang("Unable to synchronize bibliography"), JOptionPane.ERROR_MESSAGE);
                    LOGGER.debug("BibEntry not found", ex);
                } catch (Exception ex) {
                    LOGGER.warn("Could not update bibliography", ex);
                }
```

which probably is slightly better.

The main "problem" is that there is very limited exception handling locally in the OO/LO part. Everything is propagated to the calling function. Which is good from some perspectives but makes it hard to change some things as you do not know what is actually happening and how it affect things. (Unless you really understand the code...)